pkgs/by-name/sw/swagger-typescript-api/package.nix

```diff
@@ -0,0 +1,63 @@
+{
+  lib,
+  mkYarnPackage,
+  fetchFromGitHub,
+  fetchYarnDeps,
+  srcOnly,
+  stdenvNoCC,
+}:
+let
+  pname = "swagger-typescript-api";
+  version = "13.0.23";
+
+  src = srcOnly {
+    name = "${pname}-${version}"; # -source appended by srcOnly
+
+    stdenv = stdenvNoCC;
+
+    src = fetchFromGitHub {
+      owner = "acacode";
+      repo = "swagger-typescript-api";
+      rev = version;
+      hash = "sha256-Ahb/6RfLBJXRL0eSTDp5KPBdtwisCnNnugJ09LsOUTA=";
+    };
+
+    # Project use corepack and modern yarn (v4+), which has different lockfile format than yarn v1.
+    # Use yarn/lockfile v1 for usage with mkYarnPackage and fetchYarnDeps.
+    patches = [ ./yarnv1.patch ];
```

Comment:
Have you tried the new yarn-berry fetcher? https://nixos.org/manual/nixpkgs/unstable/#javascript-yarn-v3-v4